src/main/telemetry/smartport.c

```diff
@@ -303,13 +557,31 @@ void handleSmartPortTelemetry(void)
         return;
     }
 
+    if(smartPortFrameReceived) {
+        smartPortFrameReceived = false;
+        // do not check the physical ID here again
+        // unless we start receiving other sensors' packets
+        smartPortFrame_t* frame = (smartPortFrame_t*)smartPortRxBuffer;
```

Comment:
This may be dangerous - `smartPortRxBuffer` has no alignment requirement (uint8_t), but `frame` contains members that must be aligned (uint16_t, uint32_t) - it may cause unaligned reads (and thus UsageFault, killing the quad)
`smartPortFrame_t smartPortRxBuffer` will solve this